src/dags/importscripts/import_processenverbaalverkiezingen.py

```diff
@@ -0,0 +1,134 @@
+import csv
+import ftplib
+from more_ds.network.url import URL
+from os import path as ospath
+from typing import List, TypedDict, Tuple, Optional, Generator
+
+
+class Data(TypedDict):
+    """Datastructure processenverbaal verkiezingen"""
+
+    verkiezingsjaar: int
+    volgnummer: str
+    uri: str
+    documentnaam: str
+    stemlocatie: str
+    id: str
+
+
+class FTPListing:
+    """This class is used for traversing a given folder base on a FTP
+    server (objectstore) using BFS (Breadth-First search) algorithm.
+    """
+
+    def __init__(self, connection: ftplib.FTP) -> None:
+        """Initialize
+
+        Args:
+            Connection: Hold the connection to the objectstore
+
+        """
+        self.connection = connection
+
+    def list_dirs_and_files(self, _path: str) -> Tuple[List[str], Optional[List[str]]]:
+        """Return files and directories within a path
+        So it can be used to identifiy if there is need for another directory scan
+        to locate files. Files already found can be used to get the URI.
+
+        Args:
+            _path: The path for looking for files and directories (within the same path)
+
+        Returns:
+            list of found dictories and files
+
+        """
+
+        list_, dirs, files = [], [], []
+        try:
+            self.connection.cwd(_path)
+        except Exception:
+            return [], []
+        else:
+            self.connection.retrlines("LIST", lambda x: list_.append(x.split()))
+            for info in list_:
+                type, name = info[0], info[-1]
+                if type.startswith("d"):
+                    dirs.append(name)
+                else:
+                    files.append(name)
+            return dirs, files
+
+    def traverse_folder(self, path: str = "/") -> Generator:
+        """Recursive walk through directory tree, based on a BFS algorithm.
+        This function acts like an orchestrator for looking for files and dirs.
+
+        Args:
+            path: The path for looking for files and directories at the same level
+
+        Yields:
+            list of all files incl its path until all directories are depleted
+
+        """
+        dirs, files = self.list_dirs_and_files(path)
+        yield path, dirs, files
+        for name in dirs:
+            path = ospath.join(path, name)
+            yield from self.traverse_folder(path)
+            self.connection.cwd("..")
+            path = ospath.dirname(path)
```

Comment:
This is a pre-order traversal, not breadth-first.

Reply:
Actually it does, it gets the first level dirs, then goes to the next level etc, and ends at the leaf folders (lowest level). Like so:
In this case 2022 had no child folders. But is processed at the same time als 2021. As it's the top level.

1) ['2021', '2022']
2) ['2021_1', '2021_2']
3) ['2021_1_1']

Reply:
2022 having no children is cheating ;)

If the hierarchy has files

```
a1/a2/a3/a4
b1/b2/b3/b4
```

(and intermediate directories), then your algorithm either produces a4 before b3, or b4 before a3. BFS never does that.